packages/svelte/src/internal/client/runtime.js

```diff
@@ -790,16 +806,65 @@ export function get(signal) {
 	if ((flags & DERIVED) !== 0 && is_signal_dirty(signal)) {
 		update_derived(/** @type {import('./types.js').ComputationSignal<V>} **/ (signal), false);
 	}
+	if (DEV && is_tracing_signals && (flags & SOURCE) !== 0) {
+		const debug_source =
+			/** @type {import('./types.js').SourceSignal<V> & import('./types.js').SourceSignalDebug} */ (
+				signal
+			);
+		if (debug_source.d !== null) {
+			// eslint-disable-next-line no-console
+			console.log(debug_source.d);
+			queueMicrotask(() => {
+				debug_source.d = null;
+			});
+		}
+	}
 	return signal.v;
 }
 
+/**
+ * @template V
+ * @param {import('./types.js').Signal<V>} signal
+ * @returns {void}
+ */
+function attach_trace(signal) {
+	if (DEV && (signal.f & SOURCE) !== 0) {
+		let stack_debug = null;
+		try {
+			const error = new Error();
+			// Browser-sniffing: Chromium-based browsers have the "Error: .." line as part of the stack trace.
+			if (error.stack?.startsWith(`Error`)) {
+				// Chromium-based browsers don't do source mapping based on whether an Error object is handed to them,
+				// they purely do pattern matching on the stack trace string. This leads us to unfortunately needing to
+				// have "Error" as part of the name even if it's not really one.
+				error.stack = `ExpectedError: $log.trace\n${error.stack?.split('\n')?.at(3)}`;
+			} else {
+				// For once, Safari and Firefox do it better
+				error.name = '$log.trace';
+				error.stack = error.stack?.split('\n')?.at(2);
+			}
+			stack_debug = error;
+		} catch {
+			// Do nothing
+		}
+		if (stack_debug) {
+			const debug_source =
+				/** @type {import('./types.js').SourceSignal<V> & import('./types.js').SourceSignalDebug} */ (
+					signal
+				);
+			debug_source.d = stack_debug;
+		}
+	}
+}
+
 /**
  * @template V
  * @param {import('./types.js').Signal<V>} signal
  * @param {V} value
  * @returns {V}
  */
 export function set(signal, value) {
+	attach_trace(signal);
```

Comment:
Should `attach_trace` be called inside `set_signal_value`, rather than immediately before it? It might save us from forgetting to attach the trace in some future function that calls `set_signal_value`.

I see there are two occurrences of `set_signal_value` that _don't_ have a preceding `attach_trace`, inside `update_each_item_block` — I don't fully understand the code, but is there any real downside to always attaching the trace when a signal is set?

Reply:
The reason for it being like this is to ensure that the same depth in the stack for the place the user does the mutation.

Reply:
wouldn't we just need to just that number by 1 if `attach_trace` was inside `set_signal_value`?

Reply:
`attach_trace` is used in various places where `set_signal_value` can be the 2nd or 4th in the call-stack from memory so I tried to ensure it was always the same place. Maybe there's a more sane way of doing this.